src/NServiceBus.AcceptanceTests/Routing/When_configure_routes_for_unobtrusive_messages.cs

```diff
@@ -0,0 +1,104 @@
+                    .When(async s => await s.Send(new SomeCommand())))
+                .WithEndpoint<ReceivingEndpoint>()
+                .Done(c => c.ReceivedMessage)
+                .Run();
+
+            Assert.That(context.ReceivedMessage, Is.True);
+        }
+
+        [Test]
+        public async Task Should_use_routes_from_endpoint_mapping()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<SendingEndpointUsingEndpointMapping>(e => e
+                    .When(async s => await s.Send(new SomeCommand())))
+                .WithEndpoint<ReceivingEndpoint>()
+                .Done(c => c.ReceivedMessage)
+                .Run();
+
+            Assert.That(context.ReceivedMessage, Is.True);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool ReceivedMessage { get; set; }
+        }
+
+        public class SendingEndpointUsingRoutingApi : EndpointConfigurationBuilder
+        {
+            public SendingEndpointUsingRoutingApi()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.Conventions().DefiningCommandsAs(t => t == typeof(SomeCommand));
+
+                    var routing = new RoutingSettings(c.GetSettings());
+                    routing.RouteToEndpoint(typeof(SomeCommand).Assembly, Conventions.EndpointNamingConvention(typeof(ReceivingEndpoint)));
+                });
+            }
+        }
+
+        public class SendingEndpointUsingEndpointMapping : EndpointConfigurationBuilder
+        {
+            public SendingEndpointUsingEndpointMapping()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.Conventions().DefiningCommandsAs(t => t == typeof(SomeCommand));
+                }).AddMapping<SomeCommand>(typeof(ReceivingEndpoint));
+            }
+        }
+
+        public class ReceivingEndpoint : EndpointConfigurationBuilder
+        {
+            public ReceivingEndpoint()
+            {
+                EndpointSetup<DefaultServer>(c => c
+                    .Conventions()
+                    .DefiningCommandsAs(t => t == typeof(SomeCommand)))
+                    .IncludeType<SomeCommand>();
+            }
+
+            public class CommandHandler : IHandleMessages<SomeCommand>
+            {
+                Context testContext;
+
+                public CommandHandler(Context testContext)
+                {
+                    this.testContext = testContext;
+                }
+
+                public Task Handle(SomeCommand message, IMessageHandlerContext context)
+                {
+                    testContext.ReceivedMessage = true;
+                    return Task.FromResult(0);
+                }
+            }
+        }
+    }
+}
+
+// custom namespace is required to avoid automatically loading the type by the testing framework
+namespace CustomCommandMessageNamespace
```

Comment:
wouldn't it be enough to exclude the type from scanning explicitly like I did in i.ex. #3991. Because this would mean that you don't have to include it again in the receiving endpoint?